app/adapters/s3.py

```diff
@@ -47,4 +51,10 @@ async def download(file_name: str, folder: str) -> bytes | None:
         logging.error("Failed to download file from S3", exc_info=exc)
         return None
 
-    return await response["Body"].read()
+    body = await response["Body"].read()
+    content_type = response["ContentType"]
+
+    return {
+        "body": body,
+        "content_type": content_type,
+    }
```

Comment:
hmm checked s3 docs, the possible return fields are 
```js
{
    'Body': StreamingBody(),
    'DeleteMarker': True|False,
    'AcceptRanges': 'string',
    'Expiration': 'string',
    'Restore': 'string',
    'LastModified': datetime(2015, 1, 1),
    'ContentLength': 123,
    'ETag': 'string',
    'ChecksumCRC32': 'string',
    'ChecksumCRC32C': 'string',
    'ChecksumSHA1': 'string',
    'ChecksumSHA256': 'string',
    'MissingMeta': 123,
    'VersionId': 'string',
    'CacheControl': 'string',
    'ContentDisposition': 'string',
    'ContentEncoding': 'string',
    'ContentLanguage': 'string',
    'ContentRange': 'string',
    'ContentType': 'string',
    'Expires': datetime(2015, 1, 1),
    'WebsiteRedirectLocation': 'string',
    'ServerSideEncryption': 'AES256'|'aws:kms'|'aws:kms:dsse',
    'Metadata': {
        'string': 'string'
    },
    'SSECustomerAlgorithm': 'string',
    'SSECustomerKeyMD5': 'string',
    'SSEKMSKeyId': 'string',
    'BucketKeyEnabled': True|False,
    'StorageClass': 'STANDARD'|'REDUCED_REDUNDANCY'|'STANDARD_IA'|'ONEZONE_IA'|'INTELLIGENT_TIERING'|'GLACIER'|'DEEP_ARCHIVE'|'OUTPOSTS'|'GLACIER_IR'|'SNOW'|'EXPRESS_ONEZONE',
    'RequestCharged': 'requester',
    'ReplicationStatus': 'COMPLETE'|'PENDING'|'FAILED'|'REPLICA'|'COMPLETED',
    'PartsCount': 123,
    'TagCount': 123,
    'ObjectLockMode': 'GOVERNANCE'|'COMPLIANCE',
    'ObjectLockRetainUntilDate': datetime(2015, 1, 1),
    'ObjectLockLegalHoldStatus': 'ON'|'OFF'
}
```

no AI for now - but eventually we should aim to correctly support this - if we are going with a Response object

Reply:
yeah, i didn't think there was much reason to include all of the possible fields but if you want to then i can